readthedocs/domaindata/models.py

```diff
@@ -0,0 +1,83 @@
+"""
+Sphinx Domain modeling.
+
+http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html
+"""
+
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from django_extensions.db.models import TimeStampedModel
+
+from readthedocs.builds.models import Version
+from readthedocs.core.resolver import resolve
+from readthedocs.projects.models import Project
+from readthedocs.projects.querysets import RelatedProjectQuerySet
+
+
+class DomainData(TimeStampedModel):
+
+    """
+    Information from a project about it's Sphinx domains.
+
+    This captures data about API objects that exist in that codebase.
+    """
+
+    project = models.ForeignKey(
+        Project,
+        related_name='domain_data',
```

Comment:
Same here, we will have `project.domains` and `project.domain_data`. I think it will be clearer as `project.domains` (current relationship) and `project.sphinx_domains` (or `project.sphinxdomains`), IMO.